File: ws/arboleda/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Bird.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Abner Arboleda, Logic Legion, DCCO-ESPE
 */
public abstract class Bird extends Animal {

    public Bird(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }

}

File: hw/Vaca/u2/HW18ZooInPythonJavaScript/Zoo/ec.edu.espe.zoo.model/Bird.js
const Animal = require('./Animal');

class Bird extends Animal {
    constructor(id, description, dateOnborn, gender, numberOfBones, cage, wingSpan) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
        this.wingSpan = wingSpan;
    }

    getWingSpan() {
        return this.wingSpan;
    }

    setWingSpan(wingSpan) {
        this.wingSpan = wingSpan;
    }

    toString() {
        return `Bird{wingSpan=${this.wingSpan}${super.toString()}}`;
    }
}

module.exports = Bird;
